driver/driver.go

```diff
@@ -482,6 +471,88 @@ func (d *ContrailDriver) RevokeExternalConnectivity(req *network.RevokeExternalC
 	return nil
 }
 
+func (d *ContrailDriver) createRootNetwork() error {
+	rootNetwork, err := hns.GetHNSNetworkByName(common.RootNetworkName)
+	if err != nil {
+		return err
+	}
+	if rootNetwork == nil {
+
+		subnets := []hcsshim.Subnet{
+			{
+				AddressPrefix:  "0.0.0.0/24",
+				GatewayAddress: "0.0.0.0",
+			},
+		}
+		configuration := &hcsshim.HNSNetwork{
+			Name:               common.RootNetworkName,
+			Type:               "transparent",
+			NetworkAdapterName: d.networkAdapter,
+			Subnets:            subnets,
+		}
+		rootNetID, err := hns.CreateHNSNetwork(configuration)
+		if err != nil {
+			return err
+		}
+
+		log.Infoln("Created root HNS network:", rootNetID)
+	} else {
+		log.Infoln("Existing root HNS network found:", rootNetwork.Id)
+	}
+	return nil
+}
+
+func (d *ContrailDriver) waitForPipeToStart() error {
+	return d.waitForPipe(true)
+}
+
+func (d *ContrailDriver) waitForPipeToStop() error {
+	return d.waitForPipe(false)
+}
+
+func (d *ContrailDriver) waitForPipe(waitUntilExists bool) error {
```

Comment:
Please add some comments, what is exactly the point of this function and how it should work.